### DISCUSSION TITLE
SFM struggles with Pure forward motion — WHY?

### DISCUSSION THREAD

**Discussion by shubham-monarch** [colmap/colmap#2339] [General]
In reply to one of the [issues](https://github.com/colmap/colmap/issues/1490),  @tsattler wrote and I quote — '_Pure forward motion is problematic for structure-from-motion as it leads to instable (and degenerate) configurations._'. 

I am just trying to build an intuition regarding why that should be the case? Shouldn't it be a function of the nautre of the features present in the matching frames? 

Would the SFM pipeline still struggle if the camera were to be moving in a straight line on a road with different colored buildings on the either side?

**Comment by shubham-monarch** [comment]
Found one reason in one of the replies by @ahojnnes 
https://github.com/colmap/colmap/issues/677#issuecomment-529103950

**Reply by tsattler** [reply]
Exactly what Johannes wrote. It should also be explained in the usual textbooks on 3D vision.
